eng/packages/http-client-csharp/generator/Azure.Generator/src/Providers/ResourceProvider.cs

```diff
@@ -0,0 +1,356 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Generator.Mgmt.Models;
+using Azure.Generator.Primitives;
+using Azure.Generator.Utilities;
+using Azure.ResourceManager;
+using Microsoft.TypeSpec.Generator.ClientModel.Providers;
+using Microsoft.TypeSpec.Generator.Expressions;
+using Microsoft.TypeSpec.Generator.Input;
+using Microsoft.TypeSpec.Generator.Primitives;
+using Microsoft.TypeSpec.Generator.Providers;
+using Microsoft.TypeSpec.Generator.Snippets;
+using Microsoft.TypeSpec.Generator.Statements;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using static Microsoft.TypeSpec.Generator.Snippets.Snippet;
+
+namespace Azure.Generator.Providers
+{
+    internal class ResourceProvider : TypeProvider
+    {
+        private OperationSet _operationSet;
+        private ModelProvider _resourceData;
+        private ClientProvider _clientProvider;
+        private ModelSerializationExtensionsDefinition _modelSerializationExtensions;
+        private string _specCleanName;
+        private readonly IReadOnlyList<string> _contextualParameters;
+
+        private FieldProvider _dataField;
+        private FieldProvider _clientDiagonosticsField;
+        private FieldProvider _restClientField;
+        private FieldProvider _resourcetypeField;
+
+        public ResourceProvider(OperationSet operationSet, string specCleanName, ModelProvider resourceData, string resrouceType, ModelSerializationExtensionsDefinition modelSerializationExtensions)
```

Comment:
Why do we need the serializationExtensions here whereas we didn't need it for any SCM providers?

Reply:
Because we would like to refer to an internal field in it.
https://github.com/Azure/azure-sdk-for-net/pull/47944#discussion_r1961154011

Reply:
replace it with static invoke.